src/libstd/keyword_docs.rs

```diff
@@ -442,10 +446,72 @@ mod for_keyword { }
 ///
 /// [`match`]: keyword.match.html
 /// [`let`]: keyword.let.html
-/// [Rust book]: https://doc.rust-lang.org/stable/book/2018-edition/ch03-05-control-flow.html#if-expressions
+/// [Rust book]:
+/// https://doc.rust-lang.org/stable/book/2018-edition/ch03-05-control-flow.html#if-expressions
 /// [Reference]: https://doc.rust-lang.org/reference/expressions/if-expr.html
 mod if_keyword { }
 
+#[doc(keyword = "impl")]
+//
+/// The implementation-defining keyword.
+///
+/// The `impl` keyword is primarily used for defining implementations on types. There are two kinds
+/// of implementations: Inherent implementations and trait implementations. Inherent
+/// implementations define functions that operate on a type, known in object-oriented languages as
```

Comment:
This is not accurate; "method" in Rust simply refers to functions which may be called using dot syntax.
Inherent implementations may have functions which can't be called with dot syntax and they may also have associated consts.

The main difference between inherent and trait implementations is instead that the former is implemented directly on a type whereas in the latter case, it is implemented for (a trait, a type).